### PR TITLE
draft PR

### PR DESCRIPTION
**Pull request by vivekbisen**
Description
----
By default, `op pull-request` now opens a PR in draft state. Introduce `--skip-draft` switch to let octopolo continue using the existing functionality of creating PRs that are ready for review right away.

QA
----
- [x] Build this gem locally. `gem build octopolo.gemspec`.
- [x] Install the gem locally. `gem install --local octopolo-1.12.0.gem`
- [x] Go to a repository for creating test PRs.
- [x] Verify that you are using this version of the gem `op -v`.
- [x] Create a branch and test PR normally without any switches `op pull-request`.
  - [x] Verify that the PR is created as a draft PR.
- [x] Use `--sd` or `--skip-draft` switches with the previous command`.
  - [x] Verify that PR is created as an open PR.

### CODE REVIEW

File: lib/octopolo/commands/pull_request.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@
   c.desc "Branch to create the pull request against"
   c.flag [:d, :dest, :destination], :arg_name => "destination_branch", :default_value => Octopolo.config.deploy_branch
 
-  c.desc "Pass -dr to create this pull request as a draft"
-  c.switch [:dr, :draft], :arg_name => "draft"
+  c.desc "Pass -nd to skip creating this pull request as a draft"
```
Comment (nicksieger) [inline lib/octopolo/commands/pull_request.rb:6]: Did you mean
```suggestion
  c.desc "Pass -sd to skip creating this pull request as a draft"
```

**Comment by emmasax4** [comment]
I had no idea `--local` was a thing, that's cool. I'll use that in the future.

File: octopolo.gemspec [hunk 2]
```diff
@@ -27,9 +27,7 @@ Gem::Specification.new do |gem|
   gem.add_dependency 'nokogiri-happymapper', '~> 0.6.0' # Lock down to 0.6.x for Ruby 2.0 compatibility.
 
   gem.add_development_dependency 'rake', '~> 10.1'
-  gem.add_development_dependency 'bundler', '~> 1.16'
   gem.add_development_dependency 'rspec', '~> 2.99'
-  gem.add_development_dependency 'guard', '~> 2.6'
```
Comment (emmasax4) [inline octopolo.gemspec:32]: Why are we removing guard? I liked to use that when doing tests locally.

**Review by emmasax4 — DISMISSED** [review]
😍

File: octopolo.gemspec (hunk 2 above)
Comment (vivekbisen) [reply octopolo.gemspec:32]: `guard` is a dependency of `guard-rspec` which is installed right after this. This line isn't necessary.

**Comment by emmasax4** [comment]
Creating a draft PR works. However, no flag properly creates the PR as _not_ a draft. Also, the help screen displays this:
```
COMMAND OPTIONS
    -d, --dest, --destination=destination_branch - Branch to create the pull request against (default: main)
    -e, --[no-]editor                            - Use $EDITOR to update PR description before creating
    --[no-]sd, --[no-]skip-draft                 - Pass --sd to skip creating this pull request as a draft
    -x, --[no-]expedite                          - Pass -x to skip the prompt and infer from branch. Expects the branch to be in this format: JIRA-123_describe_pr OR JIRA_123_describe_pr

```

I think it's strange how just the skip-draft option has two dashes in the short flag version....

**Comment by vivekbisen** [comment]
I think "short" version of switch means one character. As soon as there are more, it needs two dashes.

**Comment by emmasax4** [comment]
@vivekbisen Ah okay. That seems to be what I remember, too... not a big deal. However, the flag still does not work as expected. I tried both `--sd` and `--skip-draft`.

**Review by emmasax4 — APPROVED** [review]

**Pull request merged**